### ISSUE TITLE
Change type pb when drag & drop beetween multiple trees

### ISSUE THREAD

**Issue by jlfy738** [vakata/jstree#1544]
First thanks a lot for your plugin and all your efforts.

I use jstree with dnd plugin.
I also use type plugin to set/change node icon when node are dropped.

Everything is working when I drag/drop in a single tree (with move_node event).
But it seems there is a problem if between 2 trees (with copy_node event). set_type as no effect in this case.

Here is a jsfiddle to see the pb, with some comments.
https://jsfiddle.net/julien738/qzc1zvod/

it seems it occurs if type is not known in the first tree.

**Comment by vakata** [comment]
Yes, indeed, even after copying a node and moving it (again) it will not get the star icon. Thanks for the report, I will fix this ASAP.

**Comment by vakata** [comment]
I made some changes but your example will still not work - it is an edge case. jsTree will now always apply the icon if the old node type is not available in the tree, but in your example the default node type is used, which is always available. If the type is available jstree will check if the current node type matches the current icon, and will not change the icon if the type icon and the current icon do not match (assuming custom icon). This is why your example still does not work - you have the default type in both locations (altough not explicit in the second) but with different definitions.

TLDR; if you have the same type in both instances (or `default`), make sure the definitions are the same to ensure proper icons.

**Comment by jlfy738** [comment]
Thank for your research.
I understand that the old icon (in tree1) with default type "cloud icon" was drag and drop in tree2.
Tree2 has implicit default type with different definition (no custom icon defined).
So the icon is not changed. An edge case i understand.

But in my example, there is a second step : 
After copying (and indeed at this moment, icon is not changed as you explained above), i do a set_type on copied node to change type to 'star' explicitly.

Is this another bug ?
Maybe I misunderstood what you meant ?

**Comment by jlfy738** [comment]
I have updated jsfiddle with a new "Tree 1b" where default type was replaced by a distinct type which is not in destination tree. There is no change.

**Comment by vakata** [comment]
You misunderstood what I meant - if there are types that are shared between trees (like `default`) they should have the same definition. Otherwise jstree will treat the icon as a custom one. As for an updated fiddle - share that link but make sure you use the latest source from github, as I have not released a new version yet.

**Comment by jlfy738** [comment]
Ok, I have updated jsfiddle example to latest version of jstree.
https://jsfiddle.net/julien738/qzc1zvod/

Between "Tree 1b" and "Tree 2", no types are shared.
